### ISSUE TITLE
Error R10

### ISSUE THREAD

**Issue by paulocf92** [synicalsyntax/discord.js-heroku#6]
Hello, after launching my heroku app it's giving me this log:
`2018-04-06T11:46:07.785538+00:00 heroku[web.1]: Error R10 (Boot timeout) -> Web process failed to bind to $PORT within 60 seconds of launch`
`2018-04-06T11:46:07.785690+00:00 heroku[web.1]: Stopping process with SIGKILL`
`2018-04-06T11:46:07.918173+00:00 heroku[web.1]: State changed from starting to crashed`
`2018-04-06T11:46:07.899719+00:00 heroku[web.1]: Process exited with status 137`

Is there a way to fix this?

**Comment by synicalsyntax** [comment]
Hey! Sorry for the late reply, but unless you're planning to run your bot alongside a web app, you need to add a Procfile specifying the type of dyno you want to run your bot on, which should be worker instead of web.

You can read more about it [here](https://github.com/synicalsyntax/discord.js-heroku#creating-a-procfile), and [this](https://github.com/synicalsyntax/discord.js-heroku/blob/master/Procfile) is an example Procfile you can download to move to your bot directory to use.

Let me know if you have any other questions!

**Comment by arivanandan** [comment]
If you did it without the Procfile you probably have web dynos
Scale down your web and scale up your worker dynos
heroku ps:scale web=0
heroku ps:scale worker=1

**Comment by pedroricardo** [comment]
I'm making the same mistake
```
2018-09-15T19:23:46.604159+00:00 heroku[web.1]: Error R10 (Boot timeout) -> Web process failed to bind to $PORT within 60 seconds of launch
2018-09-15T19:23:46.604228+00:00 heroku[web.1]: Stopping process with SIGKILL
2018-09-15T19:23:46.727135+00:00 heroku[web.1]: State changed from starting to crashed
2018-09-15T19:23:46.713582+00:00 heroku[web.1]: Process exited with status 137
```

**Comment by StaZarycki** [comment]
Hey, I have a problem with my dynos.
I can see only web dyno, but there's no worker dyno.
My Procfile.txt has only this:
worker: note bot.js
bot.js is main file with my bot.
Please help me ;_;

**Comment by generic-github-user** [comment]
I've been trying to fix this issue for the last couple of hours . . . for some reason, Heroku won't let me create a worker dyno, even though my `Procfile` is tracked by Git and only contains: `worker: npm start`.

![image](https://user-images.githubusercontent.com/40661852/51432028-34ebf480-1bff-11e9-8388-3014fa55c28e.png)

I'm getting this error when I try to add a worker dyno:

![image](https://user-images.githubusercontent.com/40661852/51431964-56001580-1bfe-11e9-9a74-4043ae555701.png)

I tried creating a new app but I'm getting the same issue. I also tried adding and removing the `engines` property from `package.json` and changing the command format in the `Procfile`.

It always starts out fine, then fails with these errors:

![image](https://user-images.githubusercontent.com/40661852/51432072-ac218880-1bff-11e9-9bee-848625e2804b.png)

I would really appreciate any help anyone can give.

**Comment by generic-github-user** [comment]
I've finally solved the issue . . . this was not fun. What happened was my `Procfile` was originally named `procfile`. I later renamed it to `Procfile` with the correct capitalization, but Git did not automatically detect this change, so the outdated version of the file that Heroku detected incorrectly used a `web` process as opposed to a `worker` process. Because of this, I was unable to create any workers.

To resolve the problem, I deleted the `Procfile`, committed the change, then re-created the file with the correct information and filename capitalization and committed again, as described in [this answer](https://stackoverflow.com/a/53184918).

![image](https://user-images.githubusercontent.com/40661852/51433853-7b9f1600-1c22-11e9-94f2-4ea13c2ed17f.png)

I am now able to create the correct type of process and have not received any more R10 Boot timeout errors that are caused by incorrectly configured web workers. Hope this helps someone.
